Source/Particles/Collision/MCCScattering.H

```diff
@@ -0,0 +1,73 @@
+/* Copyright 2021 Roelof Groenewald
+ *
+ * This file is part of WarpX.
+ *
+ * License: ????
+ */
+#ifndef WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+#define WARPX_PARTICLES_COLLISION_MCC_SCATTERING_H_
+
+#include "Utils/WarpXConst.H"
+
+#include <AMReX_Random.H>
+
+/** \brief Function to perform elastic scattering of a particle in the lab
+ * frame. The particle velocities transformed to the COM frame where a hard
+ * sphere collision occurs. The resulting particle velocities are transformed
+ * back to the lab frame and the input particle's velocity is updated.
+ * @param[in/out] ux, uy, uz colliding particle's velocity
+ * @param[in] uCOM_x, uCOM_y, uCOM_z velocity of the center of momentum frame.
+ * @param[in] engine random number generator.
+ */
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+void ElasticScattering ( amrex::ParticleReal& ux, amrex::ParticleReal& uy,
+                         amrex::ParticleReal& uz, amrex::ParticleReal& uCOM_x,
+                         amrex::ParticleReal& uCOM_y, amrex::ParticleReal& uCOM_z,
```

Comment:
The `uCOM` variables could be passed more cheaply as `const amrex::ParticleReal` instead of by reference.